### PR TITLE
fix: Cleans up rooms after failure to join.

### PR DESCRIPTION
**Pull request by damencho**

### CODE REVIEW

File: JitsiConferenceEventManager.js [hunk 1]
```diff
@@ -231,6 +231,10 @@ JitsiConferenceEventManager.prototype.setupChatRoomListeners = function() {
     this.chatRoomForwarder.forward(XMPPEvents.RESERVATION_ERROR,
         JitsiConferenceEvents.CONFERENCE_FAILED,
         JitsiConferenceErrors.RESERVATION_ERROR);
+    chatRoom.addListener(XMPPEvents.RESERVATION_ERROR,
+        () => {
+            conference.connection.emuc.doLeave(conference.room.roomjid);
```
Comment (shawnchin) [inline JitsiConferenceEventManager.js:236]: `conference.connection.emuc` was undefined when I tested this locally. 

Seems to work if I replace this with a rather clumsy looking `conference.connection.xmpp.connection.emuc.doLeave(conference.room.roomjid)`

**Comment by saghul** [comment]
Damencho can you elaborate on what this fixes pl?

**Comment by shawnchin** [comment]
@saghul  Apologies if I'm being nosy and speaking out of place, but I believe this PR is in response to the issue I raised here -- https://community.jitsi.org/t/ux-when-reservation-fails/110095?u=shawn

In short, room data cached on emuc was not being cleared up on room creation failure e.g on reservation or max user error. As a result, user can no longer retry the join and has to reload to recover.

**Comment by damencho** [comment]
I have edited the commit message with an explanation. And dropped the ts changes and open another PR for that.

**Review by jallamsetty1 — APPROVED** [review]

**Pull request merged**